core/src/Kokkos_MemoryTraits.hpp

```diff
@@ -81,20 +82,29 @@ using MemoryRandomAccess =
 namespace Kokkos {
 namespace Impl {
 
-static_assert((0 < int(KOKKOS_MEMORY_ALIGNMENT)) &&
-                  (0 == (int(KOKKOS_MEMORY_ALIGNMENT) &
-                         (int(KOKKOS_MEMORY_ALIGNMENT) - 1))),
-              "KOKKOS_MEMORY_ALIGNMENT must be a power of two");
-
 /** \brief Memory alignment settings
  *
  *  Sets global value for memory alignment.  Must be a power of two!
  *  Enable compatibility of views from different devices with static stride.
  *  Use compiler flag to enable overwrites.
  */
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+#ifdef KOKKOS_ENABLE_DEPRECATION_WARNINGS
+KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()
+#endif
 static constexpr unsigned MEMORY_ALIGNMENT = KOKKOS_MEMORY_ALIGNMENT;
 static constexpr unsigned MEMORY_ALIGNMENT_THRESHOLD =
     KOKKOS_MEMORY_ALIGNMENT_THRESHOLD;
+#ifdef KOKKOS_ENABLE_DEPRECATION_WARNINGS
+KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_POP()
+#endif
```

Comment:
```suggestion
static constexpr unsigned MEMORY_ALIGNMENT = KOKKOS_IMPL_MEMORY_ALIGNMENT;
static constexpr unsigned MEMORY_ALIGNMENT_THRESHOLD =
    KOKKOS_IMPL_MEMORY_ALIGNMENT_THRESHOLD;
```

Reply:
We should get rid off `KOKKOS_IMPL_MEMORY_ALIGNMENT_*` when we remove deprecated code but sure we can use it for now.